### DISCUSSION TITLE
Xiao with USB high speed interface

### DISCUSSION THREAD

**Discussion by ckrenslehner** [Seeed-Studio/OSHW-XIAO-Series#26] [Open for Vote]
Hi! Are you planning to design a Xiao which features a high speed USB interface? 

I find this interesting in terms of developing a CMSIS DAP compatible debug probe based on the Xiao form factor. 

Thanks und regards, 
Christian

**Comment by rootexpression** [comment]
I suspect the issue here is more a limitation on the software side, unless you need more than 1mb of speed.  A lot of the newer chips support usb 2.0, though the teensy 4.x boards are the only ones I know of that can make use of it (and fully utilize the bandwidth).

**Reply by ckrenslehner** [reply]
I don't really know to be honest. I will try a Xiao compatible port of a rp2040 picoprobe firmware and test the speed. :-)
Thanks

**Reply by rootexpression** [reply]
> I don't really know to be honest. I will try a Xiao compatible port of a rp2040 picoprobe firmware and test the speed. :-) Thanks

Please do let me know if you get it running at a good speed.  Its been a while since I've investigated it, and while the hardware is supposedly capable, the software didn't support it at all at that time.  I could certainly make use of it for HID devices.
